config/checkstyle-examples-checks.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+  <property name="haltOnException" value="false"/>
+
+  <property name="cacheFile" value="target/cache_xdocs_examples"/>
+
+  <module name="SuppressionFilter">
+    <property name="file" value="${checkstyle.examples-suppressions.file}"/>
+  </module>
+
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="false"/>
+  </module>
+
+  <module name="FileLength">
+    <property name="max" value="45"/>
+  </module>
+
+  <module name="RegexpOnFilename">
+    <property name="id" value="exampleFileName"/>
+    <property name="fileNamePattern" value="^Example\d+"/>
+    <property name="ignoreFileNameExtensions" value="true"/>
+    <property name="match" value="false"/>
+    <message key="regexp.filename.mismatch" value="All example files should start with Example\d+"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noTrailingWhitespace"/>
+    <property name="format" value="\s+$"/>
+    <property name="minimum" value="0"/>
+    <property name="maximum" value="0"/>
+    <property name="message" value="Trailing whitespace is not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noDefaultProperties"/>
+    <property name="format" value=".+\s=\s\(default\).*"/>
+    <property name="message" value="(default) properties are not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="xdocStartComment"/>
+    <property name="format" value="// xdoc section \-\- start"/>
```

Comment:
Require to be the only thing on the line.

Reply:
Done.